src/silo/query_engine/actions/aggregated.cpp

```diff
@@ -0,0 +1,304 @@
+) {
+   std::vector<AggregationResult> result;
+   const uint32_t result_size = limit.has_value()
+                                   ? std::min(limit.value(), static_cast<uint32_t>(tuples.size()))
+                                   : tuples.size();
+   for (size_t idx = 0; idx < result_size; idx++) {
+      std::map<std::string, std::variant<std::string, uint32_t>> fields;
+      uint32_t int_index = 0;
+      uint32_t string_index = 0;
+      for (const auto& field : field_defs) {
+         if (field.type == config::ValueType::NUMBER) {
+            fields[field.name] = tuples[idx].int_columns[int_index++];
+         } else if (field.type == config::ValueType::DATE) {
+            fields[field.name] = silo::common::dateToString(tuples[idx].int_columns[int_index++]);
+         } else {
+            fields[field.name] = tuples[idx].string_columns[string_index++];
+         }
+      }
+      result.push_back({fields});
+   }
+   return result;
+}
+
+}  // namespace silo::query_engine::actions
+
+template <>
+struct std::hash<silo::query_engine::actions::Tuple> {
+   std::size_t operator()(const silo::query_engine::actions::Tuple& tuple) const {
+      std::size_t seed = tuple.string_columns.size();
+      for (const std::string& value : tuple.string_columns) {
+         auto hash = std::hash<std::string>()(value);
+         // NOLINTBEGIN(readability-magic-numbers)
+         hash = ((hash >> 16) ^ hash) * 0x45d9f3b;
+         hash = ((hash >> 16) ^ hash) * 0x45d9f3b;
+         hash = (hash >> 16) ^ hash;
+         seed ^= hash + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+         // NOLINTEND(readability-magic-numbers)
+      }
+      return seed;
+   }
+};
+
+namespace silo::query_engine::actions {
+Aggregated::Aggregated(
+   std::vector<std::string> group_by_fields,
+   std::vector<std::string> order_by_fields,
+   std::optional<uint32_t> limit
+)
+    : group_by_fields(std::move(group_by_fields)),
+      order_by_fields(std::move(order_by_fields)),
+      limit(limit) {}
+
+QueryResult Aggregated::execute(
+   const Database& database,
+   std::vector<OperatorResult> bitmap_filters
+) const {
+   if (group_by_fields.empty()) {
+      uint32_t count = 0;
+      for (const auto& filter : bitmap_filters) {
+         count += filter->cardinality();
+      };
+      std::map<std::string, std::variant<std::string, uint32_t>> tuple_fields;
+      tuple_fields["count"] = count;
+      return QueryResult{std::vector<AggregationResult>{{tuple_fields}}};
+   }
+   if (group_by_fields.size() == 1) {
+      const std::string group_by_field = group_by_fields[0];
+      const auto& metadata = database.database_config.getMetadata(group_by_field);
+      if (metadata.name == database.database_config.schema.primary_key) {
+         throw QueryParseException("Cannot group by primary key field: '" + group_by_field + "'");
+      }
+      // TODO(taepper) optimize when equal to partition_by field
+      // TODO(taepper) optimize single field groupby
```

Comment:
Please add an issue number instead of a name. Experience shows that TODOs without ticket numbers are very likely to stay in the code forever.